client/reader/full-post/main.jsx

```diff
@@ -346,7 +346,7 @@ FullPostDialog = React.createClass( {
 			}
 
 			if ( shouldShowComments ) {
-				buttons.push( <CommentButton key="comment-button" commentCount={ this.props.commentCount } onClick={ this.handleCommentButtonClick } tagName="div" /> );
+				buttons.push( <CommentButton key="comment-button" commentCount={ this.props.commentCount } onClick={ this.handleCommentButtonClick } tagName="div" postId={ post.ID } siteId={ post.site_ID } /> );
```

Comment:
The result of `getPostTotalCommentsCount()` isn't always defined, so we need another way to get at the comment count. I explained the reason for passing in `commentCount` instead of fetching the total comments count in CommentButton in my [reply above](https://github.com/Automattic/wp-calypso/pull/7581#discussion_r76358978).